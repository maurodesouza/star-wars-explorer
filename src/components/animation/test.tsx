import { screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { Animation, AnimationProps } from '.';

jest.mock('react-lottie', () => ({
  __esModule: true,
  default: function Mock() {
    return <svg data-testid="lottie" />;
  },
}));

const dataMock: AnimationProps = {
  animation: 'loading',
  label: 'foo',
};

describe('<Animation />', () => {
  it('should render the animation styles correct', () => {
    const { container } = renderWithProviders(<Animation {...dataMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the animation component correct', () => {
    renderWithProviders(<Animation {...dataMock} />);

    const label = screen.getByText(dataMock.label!);
    const lottie = screen.getByTestId('lottie');

    expect(label).toBeInTheDocument();
    expect(lottie).toBeInTheDocument();
  });
});
