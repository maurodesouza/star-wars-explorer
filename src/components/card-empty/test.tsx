import { screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { EmptyCard } from '.';

const dataMock = {
  id: 'test',
};

describe('<EmptyCard />', () => {
  it('should render the empty card styles correct', () => {
    const { container } = renderWithProviders(<EmptyCard {...dataMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the empty card component correct', () => {
    renderWithProviders(<EmptyCard {...dataMock} />);

    const label = screen.getByText(`${dataMock.id}#`);

    expect(label).toBeInTheDocument();
  });
});
