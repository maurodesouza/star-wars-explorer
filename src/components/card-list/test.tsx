import { cleanup } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { Entities } from 'types';
import { createArray } from 'utils';

import { CardList } from '.';

jest.mock('components', () => ({
  __esModule: true,
  Card: function Mock() {
    return <div data-testid="card" />;
  },
}));

const itemMock = {
  id: '1',
  title: 'C-3PO',
  image: 'image',
  entity: Entities.CHARACTERS,
};

describe('<CardList />', () => {
  describe('styles', () => {
    beforeAll(() => {
      jest.resetAllMocks();
    });

    it('should render the card list styles correct', () => {
      const { container } = renderWithProviders(<CardList />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should render the items card correct', () => {
    const counts = [4, 8];

    counts.forEach(count => {
      const items = createArray(count).map((_, index) => ({
        ...itemMock,
        id: String(index),
      }));

      const render = renderWithProviders(<CardList items={items} />);
      const findedItems = render.getAllByTestId(/card/i);

      expect(findedItems.length).toBe(count);

      cleanup();
    });
  });

  it('should render the fills correct', () => {
    const minCountToFill = 4;
    const counts = [1, 3, 4, 6];

    counts.forEach(count => {
      const result = count - minCountToFill;
      const lackForFill = result <= 0 ? Math.abs(result) : 0;

      const items = createArray(count).map((_, index) => ({
        ...itemMock,
        id: String(index),
      }));

      const { container } = renderWithProviders(<CardList items={items} />);
      const findedItems = container.firstElementChild!.querySelectorAll(
        'div:not([data-testId]'
      );

      expect(findedItems.length).toBe(lackForFill);
      cleanup();
    });
  });
});
