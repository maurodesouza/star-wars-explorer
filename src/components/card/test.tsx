/* eslint-disable @typescript-eslint/no-var-requires */

import { act, fireEvent, screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { Entities } from 'types';
import { Card } from '.';

const useRouter = jest.spyOn(require('next/router'), 'useRouter');
const push = jest.fn();

useRouter.mockImplementation(() => ({
  push,
}));

const mockData = {
  id: '1',
  label: 'C-3PO',
  image:
    'https://raw.githubusercontent.com/tbone849/star-wars-guide/master/build/assets/img/characters/2.jpg',
};

describe('<Card />', () => {
  it('should render the card correct', () => {
    renderWithProviders(<Card {...mockData} />);

    expect(
      screen.getByRole('img', { name: mockData.label })
    ).toBeInTheDocument();

    expect(
      screen.getByRole('heading', { name: mockData.label })
    ).toBeInTheDocument();
  });

  it("should render the no image found element when can't load image url", async () => {
    renderWithProviders(<Card {...mockData} />);

    const image = screen.getByRole('img', { name: mockData.label });

    act(() => {
      fireEvent.error(image);
    });

    const noImageFoundElement = screen.getByText(/no found/i);

    expect(noImageFoundElement).toBeInTheDocument();
  });

  it('should render the card styles correct', () => {
    const { container } = renderWithProviders(<Card {...mockData} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should navigate to entity page', () => {
    const { container } = renderWithProviders(<Card {...mockData} />);

    fireEvent.click(container.firstChild!);

    expect(push).toBeCalledTimes(1);
    expect(push).toBeCalledWith(`/${Entities.CHARACTERS}/${mockData.label}`);
  });
});
