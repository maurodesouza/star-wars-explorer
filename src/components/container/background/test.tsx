import { renderWithProviders } from 'utils/test/helpers';
import { Background } from '.';

describe('components/container', () => {
  describe('<Background />', () => {
    it('should render the background styles correct', () => {
      const { container } = renderWithProviders(<Background />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
