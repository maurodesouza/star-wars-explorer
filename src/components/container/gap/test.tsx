import { renderWithProviders } from 'utils/test/helpers';
import { Gap } from '.';

describe('components/container', () => {
  describe('<Gap />', () => {
    it('should render the gap styles correct', () => {
      const { container } = renderWithProviders(<Gap />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
