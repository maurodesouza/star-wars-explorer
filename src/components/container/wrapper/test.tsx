import { renderWithProviders } from 'utils/test/helpers';
import { Wrapper } from '.';

describe('components/container', () => {
  describe('<Wrapper />', () => {
    it('should render the wrapper styles correct', () => {
      const { container } = renderWithProviders(<Wrapper />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
