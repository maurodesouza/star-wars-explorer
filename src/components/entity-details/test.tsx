import { screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { Entities } from 'types';

import { mock } from './mock';
import { EntityDetails, EntityDetailsProps } from '.';

const dataMock: EntityDetailsProps = {
  ...mock,
  id: '1',
  title: 'label',
  entity: Entities.CHARACTERS,
};

describe('<EntityDetails />', () => {
  it('should render the entity details styles correct', () => {
    const { container } = renderWithProviders(<EntityDetails {...dataMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the entity details component correct', () => {
    renderWithProviders(<EntityDetails {...dataMock} />);

    const image = screen.getByRole('img', { name: dataMock.title });

    expect(image).toBeInTheDocument();

    Object.entries(mock.extras)
      .reduce((arr, [key, value]) => [...arr, `${key}:`, value], [] as string[])
      .forEach(text => {
        const el = screen.getByText(text);

        expect(el).toBeInTheDocument();
      });
  });

  it('should no render content if no extras was passed', () => {
    const { container } = renderWithProviders(
      <EntityDetails {...dataMock} extras={undefined} />
    );

    const component = container.children[0];
    const content = component.children[1];

    expect(content.children.length).toBe(0);
  });
});
