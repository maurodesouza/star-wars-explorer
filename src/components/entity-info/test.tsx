import { screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import EntityDetails from '.';

describe('<EntityDetails />', () => {
  it('should render the heading', () => {
    renderWithProviders(<EntityDetails />);

    expect(screen.getByRole('heading', { name: /EntityDetails/i })).toBeInTheDocument();
  });
});
