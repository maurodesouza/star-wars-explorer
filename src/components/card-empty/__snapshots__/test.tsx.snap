// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyCard /> should render the empty card styles correct 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  width: 100%;
  height: 26rem;
  background-color: #FFF;
  color: #000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  position: relative;
  border-radius: 0.8rem;
  gap: 1.2rem;
  border: 1px solid #FFF;
}

.c2 {
  font-size: 2.4rem;
}

<div
  class="c0"
>
  <svg
    aria-hidden="true"
    class="c1"
    fill="currentColor"
    focusable="false"
    height="40"
    viewBox="0 0 384 512"
    width="40"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
      fill="currentColor"
    />
  </svg>
  <strong
    class="c2"
  >
    test
    #
  </strong>
</div>
`;
